File: Lab9/Question1/src/Door.java
public class Door {
    private boolean isOpen;

    public Door() {
        isOpen = false;
    }

    public boolean isOpen() {
        return isOpen;
    }

    public void open() {
        isOpen = true;
    }

    public void close() {
        isOpen = false;
    }

    public static void testDoor() {
        Door boyNext = new Door();
        System.out.println(boyNext.isOpen() == false);
        boyNext.open();
        System.out.println(boyNext.isOpen() == true);
        boyNext.close();
        System.out.println(boyNext.isOpen() == false);
    }
}

File: Lab9/Door.kt
class Door {
    private var isOpen: Boolean

    constructor() {
        this.isOpen = false
    }

    fun isOpen(): Boolean {
        return isOpen
    }

    fun open() {
        isOpen = true
    }

    fun close() {
        isOpen = false
    }

    companion object {
        fun testDoor() {
            val boyNext = Door()
            println(boyNext.isOpen() == false)
            boyNext.open()
            println(boyNext.isOpen() == true)
            boyNext.close()
            println(boyNext.isOpen() == false)
        }
    }
}